src/manifolds/ProbabilitySimplexEuclideanMetric.jl

```diff
@@ -20,37 +20,6 @@ function manifold_volume(M::MetricManifold{ℝ,<:ProbabilitySimplex,<:EuclideanM
     return sqrt(n + 1) / factorial(n)
 end
 
-@doc raw"""
-    rand(::MetricManifold{ℝ,<:ProbabilitySimplex,<:EuclideanMetric}; vector_at=nothing, σ::Real=1.0)
-
-
-When `vector_at` is `nothing`, return a random (uniform) point `x` on the [`ProbabilitySimplex`](@ref) with the Euclidean metric
-manifold `M` by normalizing independent exponential draws to unit sum, see [Devroye:1986](@cite), Theorems 2.1 and 2.2 on p. 207 and 208, respectively.
-
-When `vector_at` is not `nothing`, return a (Gaussian) random vector from the tangent space
-``T_{p}\mathrm{\Delta}^n``by shifting a multivariate Gaussian with standard deviation `σ`
-to have a zero component sum.
-
-"""
-rand(::MetricManifold{ℝ,<:ProbabilitySimplex,<:EuclideanMetric}; σ::Real=1.0)
-
-function Random.rand!(
```

Comment:
Why did you remove this?

Reply:
it is no longer covered by tests so I checked and it is now passed to the simplex itself. The alternative is to check whether this exact call is still valid, but then we would have to come up with a new test for code lines that we did not actively change.

Reply:
This is bad then. The old calls should use this deleted method. If they don't then we perhaps need to manually tweak dispatch for simplex.

Reply:
I See. I hope that at least one of my next PRs will not turn into an infinite round of discussions as this one does now as well.

I think this is not bad, since I am even more surprised that rand depends on a metric. But sure. then this PR will stay open for much longer until this method is back and has test coverage. The main problem then is, that I did not write this, so it will take some time to get into all the ideas behind a Euclidean metric here, since I think this is not even a manifold. but well.

Reply:
For me at least something like this was expected and I consider that code duplication reduction.
The tests still produce valid random points and vectors even for the Euclidean metric, though.

Reply:
The code that is called now is

https://github.com/JuliaManifolds/Manifolds.jl/blob/e46a864447274646f75fb1e79c5492770fd05a8f/src/manifolds/ProbabilitySimplex.jl#L377-L394

which for the points has an additional abs2 for numerical safety, the vectors have a `change_metric` additionally to the code here, but since the metric here is Euclidean, change_metric is the identity. So that code does the same.

Reply:
No, that was not correct. But I thinking just had the right idea; we use base_manifold in the new dispatch version; that unwraps "too much" we should just remove one decorator. Changing that should make this version necessary again. Will check tomorrow.

Reply:
The difference is that changing a metric may change the preferred distribution from which we draw random points and tangent vectors. I will see if I can fix that.

Reply:
Yes that is what I noticed as well then, the one with changing the metric is not called. I am a bit surprised that these 2 are affected by the change, but I can check later today as well how that happens – you are right it should not.

Reply:
I think I more or less fixed it but actually I'm not sure if the dispatch was entirely correct before this PR. It looks like it was when `rng` was provided but otherwise not. I don't really like having to define those effectively repeated methods but I don't see a better way.

Reply:
Yes, it's not ideal but `rand` is kind of an exception to our trait rules. Technically we don't have to block it propagating through `MetricManifold` because we still get a correct point, so it gets propagated by default and stuff happens in wrong order.